Libraries/StyleSheet/ColorPropType.js

```diff
@@ -9,26 +9,31 @@
   * @providesModule ColorPropType
   */
 'use strict';
+
 var ReactPropTypes = require('ReactPropTypes');
+
 var tinycolor = require('tinycolor');
 
-var colorValidator = function (props, propName) {
+var ColorPropType = function(props, propName) {
   var selectedColor = props[propName];
-  if (selectedColor === null || selectedColor === undefined || selectedColor.toString().trim() === '') {
-    return new Error(
-      `Invalid argument supplied to ${propName}.Expected a string like #123ADF or 'red'.`
-    );
+  if (selectedColor === undefined) {
+    return;
   }
 
-  if (tinycolor(selectedColor.toString().trim()).isValid()) {
-    return null;
+  if (tinycolor(selectedColor) === null) {
+    return new Error(
+`Invalid color supplied to ${propName}: ${selectedColor}. Valid color formats are
+  - '#f0f' (#rgb)
+  - '#ff00ff' (#rrggbb)
+  - '#ff00ff00' (#aarrggbb)
```

Comment:
The user facing int32 will be tricky to swap. Do you know why do we even expose it? Are there a lot of use cases for it? I have no idea how to grep the codebase for it.

One potential way forward would be to just stop supporting it. It doesn't work on the web and hasn't been very well documented anyway, I don't expect that many call sites to break.